add some error handling

gareth5james opened the pull request.

gareth5james committed:
add some error handling

roisinoreilly commented on nc-games/src/Components/ReviewsByCategory.jsx:
```diff
@@ -7,17 +7,21 @@ function ReviewsByCategory({sortBy, setSortBy, orderBy, setOrderBy}) {
             setItems(items)
             setIsLoading(false);
-        })
+        }).catch(e => {
```
you might want to change this slightly so that in the event of a network error it doesn't display your valid categories as invalid

roisinoreilly reviewed (approved):
really great job your app is really coming together, I have a couple of pointers

- at the moment if the network is throttled and you click on one of the category links we get the error message that there is no such category which isn't quite right
- the app attempts to render part of your single review page then displays your loading screen for a valid non existent review id
- you've error handled for an empty comment body but not really set strict parameters for what is allowed in a comment, eg. the user can successfully post a comment with just a space, it might be nice to set some form validation
- it would be nice to be able to undo votes, at the moment even if we try and downvote after upvoting we still get the same error message as if we're voting up twice

feel free to submit another PR if you really want but overall your error handling is great so I'm happy to approve it and let you make any changes yourself. Since you're almost finished with a lot of time to spare I'd suggest working on styling, logging in and out as different users, and implementing any of the additional tasks from your be api if you did any :)

gareth5james commented:
- struggling to duplicate the error with "no such category"
- fixed
- not sure on how to do this, will come back to
- not sure on how to do this, will come back to

added some custom error handling

gareth5james committed:
some more error handling
